homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -75,6 +85,50 @@ def get_platforms(config_entry):
     return []
 
 
+async def async_create_miio_device_and_coordinator(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up a data coordinator and one miio device to service multiple entities."""
+    model = entry.data[CONF_MODEL]
+    host = entry.data[CONF_HOST]
+    token = entry.data[CONF_TOKEN]
+    name = entry.title
+    device = None
+    if model not in MODELS_HUMIDIFIER:
+        return
+    if model in MODELS_HUMIDIFIER_MIOT:
+        device = AirHumidifierMiot(host, token)
+    else:
+        device = AirHumidifier(host, token, model=model)
```

Comment:
I suggest removing the fan entity for humidifiers.

```suggestion
    if model in MODELS_HUMIDIFIER_MIOT:
        device = AirHumidifierMiot(host, token)
    else:
        device = AirHumidifier(host, token, model=model)
        
    # removing fan platform entity for humidifiers
    entity_registry = await er.async_get_registry(hass)
    entity_id = entity_registry.async_get_entity_id("fan", DOMAIN, entry.unique_id)
    if entity_id:
        entity_registry.async_remove(entity_id)
```

Reply:
`They can simple disable not used entities.` removed from PR description. Entity registry import added.